src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripMenuItem.cs

```diff
@@ -300,11 +300,7 @@ public bool CheckOnClick
     [SRDescription(nameof(SR.CheckBoxCheckStateDescr))]
     public CheckState CheckState
     {
-        get
-        {
-            object checkState = Properties.GetInteger(s_propCheckState, out bool found);
-            return (found) ? (CheckState)checkState : CheckState.Unchecked;
-        }
+        get => Properties.GetValueOrDefault<CheckState>(s_propCheckState);
```

Comment:
It is clearer now that we have the parameter to show the default value in this case. (Then you don't need to know Unchecked is the default)

```suggestion
        get => Properties.GetValueOrDefault(s_propCheckState, CheckState.Unchecked);
```